daemon/oci_linux.go

```diff
@@ -230,6 +230,18 @@ func delNamespace(s *specs.Spec, nsType specs.NamespaceType) {
 }
 
 func setNamespaces(daemon *Daemon, s *specs.Spec, c *container.Container) error {
+	userNS := false
+	// user
+	if c.HostConfig.UsernsMode.IsPrivate() {
+		uidMap, gidMap := daemon.GetUIDGIDMaps()
+		if uidMap != nil {
+			userNS = true
+			ns := specs.Namespace{Type: "user"}
+			setNamespace(s, ns)
+			s.Linux.UIDMappings = specMapping(uidMap)
+			s.Linux.GIDMappings = specMapping(gidMap)
+		}
+	}
```

Comment:
This change is actually more minor than it appears--this is the exact block from the end of the function moved to the top to remove any need for special code at the end to determine if the userns ended up being shared or will be a newly created one during `clone()`. (because `setNamespace` does the proper job of replacement if called below)